src/App.tsx

```
@@ -1,27 +1,71 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  const handleErrors = () => {
+    setIsLoading(true);
+    getAll()
+      .then((data: Good[]) => {
+        setGoods(data);
+      })
+      .catch(() => setError('Try again later'))
+      .finally(() => setIsLoading(false));
+  };
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  return (
+    <div className="App">
+      <h1>Dynamic list of Goods</h1>
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+      {isLoading && <p> ... </p>}
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+      {!isLoading && (
+        <>
+          <button
+            onClick={() => {
+              handleErrors();
+            }}
+            type="button"
+            data-cy="all-button"
+          >
+            Load all goods
+          </button>
 
-    <GoodsList goods={[]} />
-  </div>
-);
+          <button
+            onClick={() => {
+              get5First().then((data: Good[]) => {
+                setGoods(data);
+              });
```

This violates checklist item #3: "[API] - When you make a request to the server, don't forget to handle possible errors". The handler that calls `get5First()` does not handle errors or update `isLoading`. If the request fails the UI will silently do nothing and the user gets no feedback. Make this flow consistent with the `getAll()` flow: set `isLoading` true at start, clear previous `error`, call the API, set goods on success, set `error` on failure, and ensure `isLoading` is set false in finally (or the `.finally()` callback).